crates/db_connection_pool/src/postgrespool.rs

```diff
@@ -110,8 +117,9 @@ impl PostgresConnectionPool {
                 }
                 if let Some(pg_sslmode) = params.get("pg_sslmode") {
                     match pg_sslmode.to_lowercase().as_str() {
-                        "disable" | "require" | "prefer" | "verify-ca" | "verify-full" => {
```

Comment:
`verify-ca` and `verify-full` not supported by tokio_postgres, but we simulate it by controlling hostname and cert validation, check this lines
https://github.com/spiceai/spiceai/blob/1bdb613457787c938388948b10ac443df89f5526/crates/db_connection_pool/src/postgrespool.rs#L203-L206